solana/programs/matching-engine/src/fallback/processor/initialise_fast_market_order.rs

```diff
@@ -0,0 +1,199 @@
+    let fast_market_order_vaa_digest = fast_market_order.digest();
+    let fast_market_order_vaa_digest_hash =
+        keccak::Hash::try_from_slice(&fast_market_order_vaa_digest).unwrap();
+    let verify_hash_data =
+        VerifyHashData::new(guardian_set_bump, fast_market_order_vaa_digest_hash);
+    let verify_hash_shim_ix = VerifyHash {
+        program_id: &wormhole_svm_definitions::solana::VERIFY_VAA_SHIM_PROGRAM_ID,
+        accounts: VerifyHashAccounts {
+            guardian_set: &guardian_set.key(),
+            guardian_signatures: &guardian_set_signatures.key(),
+        },
+        data: verify_hash_data,
+    }
+    .instruction();
+    // Make the cpi call to verify the shim.
+    invoke_signed_unchecked(&verify_hash_shim_ix, accounts, &[])?;
+    // ------------------------------------------------------------------------------------------------
+    // End of cpi call to verify the shim.
+
+    // Start of fast market order account creation
+    // ------------------------------------------------------------------------------------------------
+    let fast_market_order_key = fast_market_order_account.key();
+    let space = 8_usize.saturating_add(std::mem::size_of::<FastMarketOrderState>());
+    let (fast_market_order_pda, fast_market_order_bump) = Pubkey::find_program_address(
+        &[
+            FastMarketOrderState::SEED_PREFIX,
+            fast_market_order_vaa_digest.as_ref(),
+            fast_market_order.close_account_refund_recipient.as_ref(),
+        ],
+        &ID,
+    );
+
+    if fast_market_order_pda != fast_market_order_key {
+        msg!("Fast market order pda is invalid");
+        return Err(MatchingEngineError::InvalidPda.into())
+            .map_err(|e: Error| e.with_pubkeys((fast_market_order_key, fast_market_order_pda)));
+    }
+    let fast_market_order_seeds = [
+        FastMarketOrderState::SEED_PREFIX,
+        fast_market_order_vaa_digest.as_ref(),
+        fast_market_order.close_account_refund_recipient.as_ref(),
+        &[fast_market_order_bump],
+    ];
+    let fast_market_order_signer_seeds = &[&fast_market_order_seeds[..]];
+    // Create the account using the system program. The create account reliably ensures that the account creation cannot be raced.
+    create_account_reliably(
+        &signer.key(),
+        &fast_market_order_key,
+        fast_market_order_account.lamports(),
+        space,
+        accounts,
+        &ID,
+        fast_market_order_signer_seeds,
+    )?;
+    // Borrow the account data mutably
+    let mut fast_market_order_account_data = fast_market_order_account.try_borrow_mut_data()?;
+
+    // Write the discriminator to the first 8 bytes
+    let discriminator = FastMarketOrderState::discriminator();
+    fast_market_order_account_data[0..8].copy_from_slice(&discriminator);
+
+    let fast_market_order_bytes = bytemuck::bytes_of(&data.fast_market_order);
```

Comment:
With the comment above, should only have to pass in `fast_market_order` since it will be a ref.